.github/workflows/hardware-long.yml

```diff
@@ -136,69 +137,97 @@ jobs:
             --tag e2e \
             --hardware-map /opt/tenstorrent/twister/hw-map.yml --west-flash \
             -T ../tt-zephyr-platforms/app -ll DEBUG
-
-
-      - name: Fix APT URLs
+      - name: Install TT-SMI and reset the device
         run: |
-          # Patch APT URLs to use https
           sed -i -e 's/http/https/g' /etc/apt/sources.list
           apt update
-
-      - name: Install clang 17
-        run: |
-          wget https://apt.llvm.org/llvm.sh
-          sed -i "s/http:\/\/apt.llvm.org/https:\/\/apt.llvm.org/" -i llvm.sh
-          chmod +x llvm.sh
-          ./llvm.sh 17 all
-          rm -f llvm.sh
-
+          apt install -y cargo
+          pip install git+https://github.com/tenstorrent/tt-smi@v3.0.5
+          tt-smi -r 0
+  hardware-metal-test:
+    needs:
+      - hardware-zephyr-test
```

Comment:
indentation looks wrong here.

Reply:
Ah sorry - it's not the indentation, it's that another job was defined.

All steps should all be part of the same job.